File: 0740-delete-and-earn/0740-delete-and-earn.py
class Solution:
    def deleteAndEarn(self, nums: List[int]) -> int:
        nums = Counter(nums)
        arr = [0 for i in range(max(nums) + 1)]
        arr[1] = nums[1]
        for i in range(2,len(arr)):
            arr[i] += max(nums[i] * i + arr[i - 2],arr[i - 1])
        return max(arr)
        
        

File: 0740-delete-and-earn/0740-delete-and-earn.cpp
class Solution {
public:
    int deleteAndEarn(vector<int>& nums) {
        vector<int> dp(10000+1);
        for(int num:  nums) {
            dp[num] += num;
        }
        
        
        int ans = 0;
        for(int i = 3; i < dp.size(); i++) {
            dp[i] += max(dp[i-2], dp[i-3]);
            ans = max(ans, dp[i]);
        }
        
        return ans;
        
    }
};